pageserver/src/tenant.rs

```diff
@@ -289,6 +290,9 @@ pub struct Tenant {
     /// **Lock order**: if acquring both, acquire`timelines` before `timelines_creating`
     timelines_creating: std::sync::Mutex<HashSet<TimelineId>>,
 
+    /// Possibly offloaded and archived timelines
+    timelines_offloaded: Mutex<HashMap<TimelineId, Arc<OffloadedTimeline>>>,
```

Comment:
Is there an expected lock order between `timelines` and `timelines_offloaded` when transitioning a timeline from one to the other?  If so, would be good to put that in a comment here.

In future, we might also want to move timelines, timelines_creating and timelines_offloaded under a single lock: the reason I didn't do that earlier when adding timelines_creating was just because it would have been disruptive to all code that touches self.timelines.

Reply:
Yeah it's important to specify a lock order. I always lock `timelines` first then `timelines_offloaded`.

And yeah, ideally all three are under the same lock.

Reply:
Added a comment. The single lock idea is good for a followup I think